functions/Set-DbaRecoveryModel.ps1

```diff
@@ -0,0 +1,126 @@
+
+        .PARAMETER AllDatabases
+            This is a parameter that was included for safety, so you don't accidentally set options on all databases without specifying
+
+        .PARAMETER WhatIf
+            If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+        .PARAMETER Confirm
+            If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+        .PARAMETER EnableException
+            By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+            This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+            Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+        .NOTES
+            Tags: Recovery, RecoveryModel, Simple, Full, Bulk, BulkLogged
+            Author: Viorel Ciucu (@viorelciucu), https://www.cviorel.com
+
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+        .LINK
+            https://dbatools.io/Set-DbaRecoveryModel
+
+        .EXAMPLE
+            Set-DbaRecoveryModel -SqlInstance sql2014 -RecoveryModel Full -AllDatabases -Verbose
+            
+            Sets the Recovery Model to Full for all user databases on SQL Server instance sql2014
+            
+        .EXAMPLE
+            Set-DbaRecoveryModel -SqlInstance sql2014 -RecoveryModel BulkLogged -Database TestDB
+            
+            Sets the Recovery Model to BulkLogged for database [TestDB] on SQL Server instance sql2014
+            
+        .EXAMPLE
+            Set-DbaRecoveryModel -SqlInstance sql2014 -RecoveryModel Simple -Database TestDB1, TestDB2 -Verbose
+            
+            Sets the Recovery Model to Simple for databases [TestDB1] and [TestDB2] on SQL Server instance sql2014
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential][System.Management.Automation.CredentialAttribute()]
+        $SqlCredential,
+        [ValidateSet('Simple', 'Full', 'BulkLogged')]
+        [string]$RecoveryModel = '',
+        [Alias("Databases")]
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [switch]$AllDatabases,
+        [parameter(ValueFromPipeline = $true)]
+        [switch][Alias('Silent')]$EnableException
+    )
+    process {
+        foreach ($serverName in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $serverName -SqlCredential $SqlCredential -MinimumVersion 11
```

Comment:
This should be able to support SQL Server 2005 and higher, not just 2012.

Reply:
Also please update this try/catch block to follow the format in the template referenced earlier.